util/args.py

```diff
@@ -170,34 +165,10 @@ def get_args() -> argparse.Namespace:
         "another dataset). When not set, resnet50_inat is initalized with weights from iNaturalist2017. Other "
         "networks are initialized with weights from ImageNet",
     )
-    parser.add_argument(
-        "--disable_derivative_free_leaf_optim",
-        action="store_true",
-        help="Flag that optimizes the leafs with gradient descent when set instead of using the derivative-free "
-        "algorithm",
-    )
-    parser.add_argument(
-        "--kontschieder_train",
-        action="store_true",
-        help="Flag that first trains the leaves for one epoch, and then trains the rest of ProtoTree (instead of "
-        "interleaving leaf and other updates). Computationally more expensive.",
-    )
-    parser.add_argument(
-        "--kontschieder_normalization",
-        action="store_true",
-        help="Flag that disables softmax but uses a normalization factor to convert the leaf parameters to a "
-        "probability distribution, as done by Kontschieder et al. (2015). Will iterate over the data 10 times "
-        "to update the leaves. Computationally more expensive.",
-    )
-    parser.add_argument(
-        "--log_probabilities",
-        action="store_true",
-        help="Flag that uses log probabilities when set. Useful when getting NaN values.",
-    )
     parser.add_argument(
         "--pruning_threshold_leaves",
         type=float,
-        default=0.01,
+        default=0.055,  # TODO: Where does this value come from?

```

Comment:
As I commented somewhere in the training script, I believe, this value should not be static! It depends on the number of classes, as this is the smallest max-confidence below which leaves will be pruned. Leaves that didn't learn anything will have a distribution close to `1/num_classes * torch.ones((num_classes))`, so the threshold should rather be a multiplier that will be applied to `1/num_classes`.

It's a hyperparameter that is important to tune

Reply:
Btw, one optimization could be to find the best value automatically by some standard search algorithm. It is only applied to the already trained tree, hence the search is rather cheap

Reply:
Thanks for explaining.

I've made some changes to the PR to make this a `leaf_pruning_multiplier` arg (from which the threshold is calculated), and added some comments detailing what you explained here. Does this seem okay to you?